src/tunacode/core/session/state.py

```
@@ -154,58 +154,49 @@ def _get_session_file_path(self) -> Path:
-            serialized_messages.append(serialized_message)
+            serialized_messages.append(msg)
+
         return serialized_messages
 
-    def _deserialize_messages(self, data: list[dict]) -> list:
-        """Deserialize JSON dicts back to message objects."""
-        msg_adapter: Any | None = None
-        try:
-            from pydantic import TypeAdapter
-            from pydantic_ai.messages import ModelMessage
+    def _deserialize_messages(self, data: Any) -> list[dict[str, Any]]:
```

Ruff (0.14.14)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `data`

(ANN401)